onnxruntime/core/providers/cuda/cuda_execution_provider_info.h

```diff
@@ -45,6 +48,9 @@ struct CUDAExecutionProviderInfo {
   // arena config.
   OrtArenaCfg* default_memory_arena_cfg{nullptr};
   CUDAExecutionProviderExternalAllocatorInfo external_allocator_info{};
+  // By default use fix workspace size (32M) for Conv algo search, the final algo might not be the best.
+  // If set to true, try to use as much as possible memory for algo search.
+  bool cudnn_conv_use_max_workspace{false};
```

Comment:
I suppose we'll need a separate PR to expose this to inferencing.

Reply:
Sure. Just removed this configuration from onnxruntime_pybind_state.

Reply:
So that PR will expose this in the public facing headers/pybind ?

Reply:
I am actually not sure how to expose this to inferencing... May need Prana's or your suggestion.

Reply:
For that, you can check how other CUDA configs are getting wired from the c api. For example, you can check how users get to set `cudnn_conv_algo_search` via the `OrtCudaProviderOptions` struct in the `onnxruntime_c_api.h`. I assume, you will just have to follow the same pattern.

Reply:
cudnn_conv_algo_search defined global variable in onnxruntime_pybind_state, I did the same thing but Prana suggested we shouldn't add more global, that's the reason I removed it. I am thinking maybe we want a new way to do this.

Reply:
Yes the globals approach is deprecated. But users can provide EP specific config options via a provider options map. Please see the logic here - https://github.com/microsoft/onnxruntime/blob/e3489290193916c4e8db8867d441259d1e2cfb2e/onnxruntime/python/onnxruntime_pybind_state.cc#L504